docs/nf4_science/imaging/04_config.md

```diff
@@ -0,0 +1,464 @@
+# Part 4: Configuration
+
+In Parts 1-3, we learned how to run molkart, manage inputs with parameter files and samplesheets, and train custom segmentation models.
+Now we'll explore how to configure pipelines for different computing environments using **configuration files** and **profiles**.
+
+## Learning objectives
+
+By the end of this part, you'll be able to:
+
+- Understand how Nextflow resolves configuration from multiple sources
+- Use nf-core built-in profiles for containers and testing
+- Create custom profiles for different computing environments
+- Customize resource requests using process labels
+- Manage resource limits in constrained environments
+- Inspect resolved configuration with `nextflow config`
+
+---
+
+## 1. Understanding Nextflow configuration
+
+### 1.1. What is a configuration file?
+
+Nextflow uses configuration files to separate **workflow logic** (what to do) from **execution settings** (how and where to do it).
+
+Configuration files control:
+
+- Container engines (Docker, Singularity, Conda)
+- Compute resources (CPUs, memory, time)
+- Execution platforms (local, HPC, cloud)
+- Pipeline parameters
+
+### 1.2. Configuration precedence
+
+Nextflow loads configuration from multiple sources, with later sources overriding earlier ones:
+
+1. **Pipeline config**: `nextflow.config` in the pipeline repository
+2. **Directory config**: `nextflow.config` in your working directory
+3. **User config**: `~/.nextflow/config`
+4. **Command-line**: Parameters and options passed directly
+
+This layered approach lets you keep defaults in the pipeline, override with user-specific settings, and make quick adjustments on the command line.
+
+### 1.3. Our current configuration
+
+Let's look at the configuration we've been using:
+
+```bash
+cat nextflow.config
+```
+
+```groovy title="nextflow.config"
+docker {
+    enabled          = true
+    runOptions       = '-u $(id -u):$(id -g) --entrypoint ""'
+}
+
+process {
+    resourceLimits = [ cpus: 2, memory: 7.GB]
+}
+```
+
+This minimal config:
+
+- Enables Docker containers
+- Sets Docker run options to avoid permission issues
+- Caps resource requests at 2 CPUs and 7 GB (for GitHub Codespaces)
```

Comment:
need to define the resource requests for GitHub codespaces and what resourceLimits are.